apiserver/facades/agent/provisioner/provisioner.go

```diff
@@ -440,11 +440,19 @@ func (api *ProvisionerAPI) MachinesWithTransientErrors(ctx context.Context) (par
 		if !canAccessFunc(machine.Tag()) {
 			continue
 		}
-		if _, provisionedErr := machine.InstanceId(); provisionedErr == nil {
+		machineUUID, err := api.machineService.GetMachineUUID(ctx, coremachine.Name(machine.Id()))
+		if err != nil {
+			return results, err
+		}
+		_, err = api.machineService.InstanceID(ctx, machineUUID)
+		if err == nil {
 			// Machine may have been provisioned but machiner hasn't set the
 			// status to Started yet.
 			continue
 		}
+		if !errors.Is(err, machineerrors.NotProvisioned) {
+			return results, err
+		}
```

Comment:
**suggestion:** With the GetUUID, this whole code started to be barely readable, because we have a unexpected pattern in the middle (we call a function that we expect being in error, which is counter-intuitive, there is even a comment for that).

I think this lines (from getMachineUUID to the check against NotProvisionned) should be hidden in a unexported function with a meaningful name which explain what is going one.

Reply:
Yeah I ended up moving the instance getter out, it was verbose.